src/types/eval_const.ts

```diff
@@ -220,19 +228,16 @@ export function evalBinaryImpl(operator: string, left: Value, right: Value): Val
     }
 
     if (BINARY_OPERATOR_GROUPS.Equality.includes(operator)) {
-        if (typeof left === "string" || typeof right === "string") {
+        if (typeof left === "string" && typeof right === "string") {
             throw new EvalError(
                 `${operator} not allowed for strings ${str(left)} and ${str(right)}`
             );
         }
 
-        let isEqual: boolean;
+        const leftDec = promoteToDec(left);
+        const rightDec = promoteToDec(right);
 
-        if (left instanceof Decimal && right instanceof Decimal) {
```

Comment:
I think this change may break constant eval of `true == true`, since bool is not promotable to Decimal.

Reply:
Maybe change this to:

```
if (left instanceof Decimal || right instanceof Decimal) {
        const leftDec = promoteToDec(left);
        const rightDec = promoteToDec(right);
        
        isEqual = leftDec.equals(rightDec);
} else {
       isEqual = left === right;
}

Reply:
Good note, will fix.

Reply:
Addressed slightly different way in d06e427. Consider.